### ISSUE TITLE
5.0 P1S1+ P2S2 sum of products plugin

### ISSUE THREAD

**Issue by RichardHeenan** [SasView/sasview#1305]
External user reporting some issues in 4.2 and 5.0 trying to add two P(Q)S(Q) plugin models.

In this sasmodels branch -  **sasview-1301-mixture-results**, tried to add two plugin models, in my developer 5.0 build.

--------------------------------------------------
File   50sphxhardsphere_again.py

from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info

model_info = load_model_info('sphere*hardsphere')
model_info.name = '50sphxhardsphere_again'
model_info.description = '5.0 sph * hardsphere again'
Model = make_model_from_info(model_info)
--------------------------------------------------
File  sphere_HSSQ.py

from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info

model_info = load_model_info('sphere*hardsphere')
model_info.name = 'sphere_HSSQ'
model_info.description = '5.0 sphere * hardsphere'
Model = make_model_from_info(model_info)
--------------------------------------------------
Try to add these in add/multiple menu, get console message:

16:45:37 - INFO: Model function (sum_AB_CD) has been set!
16:46:01 - INFO: Model function (sum_AB_CD) has been set!
16:46:01 - ERROR: Error building model: No module named 'sasmodels.models.50sphxhardsphere_again'
16:46:01 - ERROR:   File "C:\sasview42\sasmodels\sasmodels\generate.py", line 963, in load_kernel_module
    __import__('sasmodels.models.'+model_name)
ModuleNotFoundError: No module named 'sasmodels.models.50sphxhardsphere_again'

--------------------------------------------------
File sum_AB_CD.py has:

from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info

model_info = load_model_info('50sphxhardsphere_again+sphere_HSSQ')
model_info.name = 'sum_AB_CD'
model_info.description = 'sum PS PS'
Model = make_model_from_info(model_info)
--------------------------------------------------
Used to have to put a custom. in to combine plugin models? Tried that, no good.

**Comment by pkienzle** [comment]
Problem may be that the filename `50sphxhardsphere_again.py` starts with a number.

**Comment by smk78** [comment]
**In 4.2.1 Release**, you can generate:
```
from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info
model_info = load_model_info('sphere*hardsphere')
model_info.name = 'dan1'
Model = make_model_from_info(model_info)
```
and
```
from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info
model_info = load_model_info('sphere*stickyhardsphere')
model_info.name = 'dan2'
Model = make_model_from_info(model_info)
```
and they are available from the plugin models drop down to use in a FitPage (as expected).

You can also combine them:
```
from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info
model_info = load_model_info('dan1+dan2')
model_info.name = 'dan3'
Model = make_model_from_info(model_info)
```
and this model is also available in the FitPage with parameters:
```
scale
background
AB_scale
A_sld
A_sld_solvent
A_radius
B_radius_effective
B_volfraction
CD_scale
C_sld
C_sld
C_sld_solvent
C_radius
D_radius_effective
D_volfraction
D_perturb
D_stickiness
```
which is what you would expect.

_NB: As @RichardHeenan notes, the docs do say that other plugins should be prefixed by custom, that is,  'custom.dan1+custom.dan2'. but this appears to be unnecessary. Is this actually redundant behaviour now, or are there consequences of not incorporating custom?_
 
If instead you copy and edit plugins dan1 & dan2 to replace * by @ to generate
```
from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info
model_info = load_model_info('sphere@hardsphere')
model_info.name = 'dan4'
Model = make_model_from_info(model_info)
```
and
```
from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info
model_info = load_model_info('sphere@stickyhardsphere')
model_info.name = 'dan5'
Model = make_model_from_info(model_info)
```
and then sum them
```
from sasmodels.core import load_model_info
from sasmodels.sasview_model import make_model_from_info
model_info = load_model_info('dan4+dan5')
model_info.name = 'dan6'
Model = make_model_from_info(model_info)
```
and load dan6 you get a FitPage with the following parameters:
```
scale
background
sld
sld_solvent
radius
volfraction
perturb
stickiness
```
The reduction in the number of parameters is presumably because several are now constrained to have the same meaning across each of the 4 models _though we do not actually present the user with any knowledge of this (but I have updated the docs for 4.2.2)_.

_However_, if I set AB_scale=CD_scale=0.5 in dan3 I do not get the same calculation as from dan6 with scale=1 (all other parameters the same), although the form factor oscillations are in the same place. To get the two models to agree I have to set AB_scale=CD_scale=0.2, which I don't understand...

**Comment by pkienzle** [comment]
> The reduction in the number of parameters is presumably because several are now constrained to have the same meaning across each of the 4 models though we do not actually present the user with any knowledge of this (but I have updated the docs for 4.2.2).

No, I believe that is just wrong.  I suspect it is happening because both `*` and `@` have a composition property, but one is a 'mixture' composition and the other is a 'product' composition.   Need to look in detail at [`make_mixture_info()`](https://github.com/SasView/sasmodels/blob/d8156b7c605bfa8ccb776825a20df21d8bf425d1/sasmodels/mixture.py#L29) to fix this.

**Comment by pkienzle** [comment]
Mixture model needs to be rewritten.  Not going to happen for 4.3/5.0.

**Comment by smk78** [comment]
So given that we are writing the Release Notes for 4.2.2 right now, I'm wondering what advice under 'Known Issues' we should give? P@S alone is ok, I believe. The issue is just if someone tries P1@S1+P2@S2?
